### DISCUSSION TITLE
Invalidating entities in SvelteKit re-execute active query in the deleting component

### DISCUSSION THREAD

**Discussion by frederikhors** [urql-graphql/urql#1927] [Q&A]
_Every time I open an issue or a question I know that I'm taking precious time away from the life of each of us. I try to open them only when it is really necessary._

I love `urql` as I love the sun and after a very boring project without GraphQL I went back to using this wonder.

Having started a new project I have left aside the [old starter](https://github.com/sveltejs/template) with [old router](https://github.com/ItalyPaleAle/svelte-spa-router) and I'm using [SvelteKit](https://kit.svelte.dev/).

Everything is working fine. _Your work is wonderful (and the world is taking notice)._

But there is one thing I noticed today implementing deleting mutations.

In the old project (same code) it worked, here it works but differently and not for the better.

- urql.ts:

```ts
const updates: GraphCacheUpdaters = {
  Mutation: {
    playerCreate: (result, args, cache) => {
      const playersQueries = cache
        .inspectFields("Query")
        .filter((x) => x.fieldName === "players");
      playersQueries.forEach(({ fieldName, arguments: variables }) =>
        cache.invalidate("Query", fieldName, variables)
      );
    },

    playerDelete: (result, args, cache) => {
      if (result.playerDelete) {
        cache.invalidate({ __typename: "Player", id: args.id });
      }
    },
  },
};
```

- playerEdit.svelte

```ts
// I need this because I'm getting player data on this component (page)

const player = operationStore<PlayerQuery>(
  PlayerDocument,
  { id: $page.params.id },
  { requestPolicy: "cache-and-network" }
);

query(player);

const playerDelete = mutation<
  PlayerDeleteMutation,
  PlayerDeleteMutationVariables
>({
  query: PlayerDeleteDocument,
  variables: { id: $page.params.id },
});

async function onDelete() {
  // player.context.pause = true; // I also tried with this

  const result = await playerDelete();

  if (result.error) {
    console.error(result.error)

    // player.context.pause = false; // I also tried with this

    return;
  }

  goto("/players", { replaceState: true });
}
```

In the old project after the `playerDelete()` I went back to the list page and the list was invalidated (with a new call for `players`), because the player was an element of that list.

Now it happens that after the `playerDelete()` a request call starts for the player I deleted too!
I suppose because SvelteKit has not yet completely destroyed the page I'm on before the invalidation takes hold and therefore the `player` query being active triggers a revalidate.

I also tried to use `player.context.pause` before the `playerDelete()` in vain.

I can't leave things like this because it's really bad to see. Especially because so many entity cancellations occur very often in this app and each `player` deleted call creates an error in the backend (for missing `player` fetch!).

Can it be a javascript event loop problem (the same micro-task)?

Here is how `goto` works: https://kit.svelte.dev/docs#modules-$app-navigation.

**Comment by frederikhors** [comment]
@kitten Any idea on this? Is there a way to "`pause`" invalidation on a query?

**Reply by frederikhors** [reply]
I feel alone here... 😢

**Reply by lakshmajee** [reply]
Hi @frederikhors I am also trying to use urql with sveltekit, Could you be able to provide me some examples on setting up. (or any example repo link)
